src/pages/Dictionary/components/DisplayMode.jsx

```diff
@@ -1,24 +1,124 @@
 import { Button } from '@nextui-org/react';
 import { LikeFilled, DislikeFilled } from '@ant-design/icons';
+import {
+  incrementLikes,
+  incrementDislikes,
+  decrementDislikes,
+  decrementLikes
+} from '../../../utils/firebaseOperations';
+import { useState } from 'react';
 
 const DisplayMode = ({ entry }) => {
+  const [likes, setLikes] = useState(entry.likes);
+  const [dislikes, setDislikes] = useState(entry.dislikes);
+  const [userAction, setUserAction] = useState(null); // 'liked', 'disliked', or null
```

Comment:
this state is for checking the user's previous action, so its not related to rendering. no need to use useState

Reply:
this is my preferred way
```
  const [like, setLike] = useState(false);
  const [dislike, setDislike] = useState(false);
  const prevAction = useRef(null);

  const handleLike = () => {
    setLike(true);
    setDislike(false);
    if (prevAction.current.value === 'dislike') {
      // this means a user clicked dislike before
      // decrement firebase dislike by 1
    }
    prevAction.current.value = 'like'
  };

  const handleDislike = () => {
    setLike(false);
    setDislike(true);
    if (prevAction.current.value === 'like') {
      // this means a user clicked like before
      // decrement firebase like by 1
    }
    prevAction.current.value = 'dislike'
  };
```

Reply:
Another reason that you might want to store the values as Ture/False is that Raina set the button background color based on if the like and dislike are true or false